docs/data-structures.md

```diff
@@ -0,0 +1,249 @@
+# Data Structures of Cell/Script/Transaction/Block
+
+This documents list all the basic structures one may need to know in order to develop on CKB, or mine CKB coins. 
+
+* [Cell](#Cell)
+* [Script](#Script)
+* [Transaction](#Transaction)
+* [Block](#Block)
+
+
+
+## Cell
+
+### Example
+
+```json
+{
+    "capacity": 5000000,
+    "data": "0x",
+    "lock": "0xa58a960b28d6e283546e38740e80142da94f88e88d5114d8dc91312b8da4765a",
+    "type": null
+}
+```
+
+## Description
+
+| Name       | Type       | Description                                                  |
+| :--------- | :--------- | :----------------------------------------------------------- |
+| `capacity` | uint64     | **The size of the cell.** When a new cell is generated (via transaction), one of the verification rule is `capacity ≥ len(capacity)+len(data)+len(type)+len(lock)`. This value also represents the balance of CKB coin, just like the `balance` field in the Bitcoin's UTXO. (E.g. Alice owns 100 CKB coins means she can unlock a group of cells that has 100 amount of `capacity` in total.) |
+| `data`     | Bytes      | **Arbitrary data.** This part is for storing states or scripts.  In order to make this cell valid on-chain, the data filled in this field should comply with the logics and rules defined by `type` or `lock`. |
```

Comment:
```suggestion
| `data`     | Bytes      | **Arbitrary data.** This part is for storing states or scripts.  In order to make this cell valid on-chain, the data filled in this field should comply with the logic and rules defined by `type`. |
```

The data field is only constrained by the `type`, The `lock` defines the ownership of the cell.

Reply:
It also could be constrained by `lock`, as `type` and `lock` both has access to all the information of the transaction.

Reply:
I see.